src/cli/util.js

```diff
@@ -12,19 +12,20 @@ const fromPairs = require("lodash/fromPairs");
 const pick = require("lodash/pick");
 const groupBy = require("lodash/groupBy");
 const flat = require("lodash/flatten");
+const prettier = require("../index");
+const thirdParty = require("../common/third-party");
+const {
+  createIgnorer,
+  errors,
+  coreOptions,
+  optionsModule,
+  optionsNormalizer,
+  utils: { arrayify, isTTY },
+} = require("./prettier-internal");
 
 const minimist = require("./minimist");
-const prettier = require("../../index");
-const createIgnorer = require("../common/create-ignorer");
 const { expandPatterns, fixWindowsSlashes } = require("./expand-patterns");
-const errors = require("../common/errors");
 const constant = require("./constant");
-const coreOptions = require("../main/core-options");
-const optionsModule = require("../main/options");
-const optionsNormalizer = require("../main/options-normalizer");
-const thirdParty = require("../common/third-party");
-const arrayify = require("../utils/arrayify");
-const isTTY = require("../utils/is-tty");
```

Comment:
Can we use https://eslint.org/docs/rules/no-restricted-modules to enforce these import rules?

Reply:
Yes, I was going to add. eslint has a good example https://github.com/eslint/eslint/blob/master/.eslintrc.js

Reply:
But we need decide this first,

How do we import `fast-glob` `lodash/flatten` `json-stable-stringify` `lodash/fromPairs` etc.

For example `lodash/fromPairs`, `index.js` might not need it, but it could be used someday in `index.js`, so all come from `index.js`?

Reply:
3rd party libraries are extremely unlikely to be stateful, so it's safe to keep them as is. I support a stricter separation of the CLI from the rest of the code, but I don't think the size of the package is important.

Reply:
On the other hand, isn't that what `third-party.js` is for?

Reply:
`third-party.js` only keep modules need mock during test.